resources/resourcebar.ts

```diff
@@ -51,33 +51,33 @@ export default class ResourceBar extends HTMLElement {
     if (!o)
       return element;
 
-    if (o.value)
+    if (typeof o.value !== 'undefined' && o.value !== null)
```

Comment:
Should these all be `typeof o.value === 'string'`?

Reply:
I think the null check is still needed as the typescript type check only ensure the type-safe when you are developing, if there is a js file that invoke this function, an explicit null-check in runtime would be better so we won't just pass null to functions that might cause potential un-expected bihaviors.

Reply:
I think if your goal is to be defensive, then `if (typeof o.value === 'string')` style might be better, because it would prevent other types as well as null?

Reply:
Sure.
I just changed these code via a regex matching without changing them one by one particularly as they are not all strings but numbers or booleans.
I'll change them manually later.

Reply:
Done.